api/v1alpha1/gateway_types.go

```diff
@@ -179,56 +182,95 @@ type ListenerTLS struct {
 
 // GatewayStatus defines the observed state of Gateway
 type GatewayStatus struct {
-	// TODO overall status
-
-	// Listeners is the status for each listener block in the
-	// Spec. The status for a given block will match the order as
-	// declared in the Spec, e.g. the status for Spec.Listeners[3]
-	// will be in Status.Listeners[3].
+	// Conditions describe the current conditions of the Gateway.
+	Conditions []GatewayCondition `json:"conditions"`
+	// Listeners provide status for each listener defined in the Spec. The name
+	// in ListenerStatus refers to the corresponding Listener of the same name.
 	Listeners []ListenerStatus `json:"listeners"`
-	// Routes is the status for each attached route to the
-	// Gateway. The status for a given route will match the orer as
-	// declared in the Spec, e.g. the status for Spec.Routes[3] will
-	// be in Status.Routes[3].
-	Routes []GatewayRouteStatus `json:"routes"`
+}
+
+// GatewayConditionType is a type of condition associated with a Gateway.
+type GatewayConditionType string
+
+const (
+	// ConditionNoSuchGatewayClass indicates that the specified GatewayClass
+	// does not exist.
+	ConditionNoSuchGatewayClass GatewayConditionType = "NoSuchGatewayClass"
+	// ConditionGatewayNotScheduled indicates that the Gateway has not been
+	// scheduled.
+	ConditionGatewayNotScheduled GatewayConditionType = "GatewayNotScheduled"
+	// ConditionListenersNotReady indicates that at least one of the specified
+	// listeners os not ready.
+	ConditionListenersNotReady GatewayConditionType = "ListenersNotReady"
+	// ConditionInvalidListeners indicates that at least one of the specified
+	// routes is invalid.
```

Comment:
s/routes/listeners/

Reply:
If the `ConditionInvalidListeners` is raised, then there must also be a corresponding entry in `.status.conditions.listeners.conditions`. We should document that here.

Reply:
Good call, thanks! I also added a similar requirement for the ListenersNotReady condition along with a new ListenerNotReady condition that can be set for specific listeners.